sdks/go/pkg/beam/core/runtime/exec/fn.go

```diff
@@ -274,52 +286,60 @@ func (n *invoker) ret2(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime,
 // ret3 handles processing of a trio of return values.
 func (n *invoker) ret3(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime, r0, r1, r2 interface{}) (*FullValue, error) {
 	switch {
-	case n.outEtIdx >= 0:
+	case n.outEtIdx == 0:
 		if n.outErrIdx == 2 {
 			if r2 != nil {
 				return nil, r2.(error)
 			}
 			n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Pane: pn}
 			return &n.ret, nil
 		}
-		if n.outPcIdx >= 0 {
+		if n.outPcIdx == 2 {
 			n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Pane: pn, Continuation: r2.(sdf.ProcessContinuation)}
 			return &n.ret, nil
 		}
 		n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Elm2: r2, Pane: pn}
 		return &n.ret, nil
-	default:
-		if n.outErrIdx == 2 {
-			if r2 != nil {
-				return nil, r2.(error)
-			}
-			if n.outPcIdx == 1 {
-				n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Pane: pn, Continuation: r1.(sdf.ProcessContinuation)}
-				return &n.ret, nil
-			}
-			n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Elm2: r1, Pane: pn}
+	case n.outErrIdx == 2:
+		if r2 != nil {
+			return nil, r2.(error)
+		}
+		if n.outPcIdx == 1 {
+			n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Pane: pn, Continuation: r1.(sdf.ProcessContinuation)}
 			return &n.ret, nil
 		}
+		n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Elm2: r1, Pane: pn}
+		return &n.ret, nil
+	default:
 		n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Elm2: r1, Pane: pn, Continuation: r2.(sdf.ProcessContinuation)}
 		return &n.ret, nil
 	}
 }
 
 // ret4 handles processing of a quad of return values.
 func (n *invoker) ret4(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime, r0, r1, r2, r3 interface{}) (*FullValue, error) {
-	if r3 != nil {
-		return nil, r3.(error)
-	}
-	if n.outEtIdx == 0 {
-		if n.outPcIdx >= 0 {
-			n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Pane: pn, Continuation: r2.(sdf.ProcessContinuation)}
+	switch {
```

Comment:
I don't really mind it too much here since its consistent with the other functions, but in general I don't love 1 condition switch statements (with or without a default) since its basically just an if. Consider changing it to an if

Reply:
That's fair, I suppose the switch format really only makes sense to leave here if we're anticipating adding another return type soon and would have to handle it, but it's a small enough change that it's easier to go with the simple option.